app/src/main/java/net/squanchy/settings/SettingsFragment.java

```diff
@@ -89,6 +105,74 @@ public void onCreate(Bundle savedInstanceState) {
         });
     }
 
+    private void onGoogleConnectionFailed() {
+        Timber.e("Google Client connection failed");
+        Snackbar.make(getViewOrThrow(), R.string.proximity_error_google_client_connection, Snackbar.LENGTH_LONG).show();
+    }
+
+    private ProximityPreconditions.Callback proximityPreconditionsCallback() {
+        return new ProximityPreconditions.Callback() {
+            @Override
+            public void notOptedIn() {
+                showProximityEnablingError(Snackbar.make(getViewOrThrow(), R.string.proximity_error_not_opted_in, Snackbar.LENGTH_LONG));
```

Comment:
I think you can encapsulate `Snackbar.make` inside `showProximityEnablingError` and pass only the message